test/grad.jl

```diff
@@ -44,6 +44,43 @@ Base.length(x::DummyType) = size(x.X, 1)
         @test Ac == A
     end
 
+    @testset "multi vars jacobian/grad" begin
+        rng, fdm = MersenneTwister(123456), central_fdm(5, 1)
+        f1(x, y) = x * y + x
+        x, y = rand(rng, 3, 3), rand(rng, 3, 3)
+        jac_xs = jacobian(fdm, f1, x, y)
+        @test jac_xs[1] ≈ jacobian(fdm, x->f1(x, y), x)
+        @test jac_xs[2] ≈ jacobian(fdm, y->f1(x, y), y)
+
+        # mixed scalar and matrices
+        x, y = rand(3, 3), 2
+        jac_xs = jacobian(fdm, f1, x, y)
+        @test jac_xs[1] ≈ jacobian(fdm, x->f1(x, y), x)
+        @test jac_xs[2] ≈ jacobian(fdm, y->f1(x, y), y)
+
+        f2(x, y) = sum(x * y + x)
+        x, y = rand(rng, 3, 3), rand(rng, 3, 3)
+        dxs = grad(fdm, f2, x, y)
+        @test dxs[1] ≈ grad(fdm, x->f2(x, y), x)
+        @test dxs[2] ≈ grad(fdm, y->f2(x, y), y)
+
+        x, y = rand(rng, 3, 3), 2
+        dxs = grad(fdm, f2, x, y)
+        @test dxs[1] ≈ grad(fdm, x->f2(x, y), x)
+        @test dxs[2] ≈ grad(fdm, y->f2(x, y), y)
+
+        f3(x::Tuple) = sum(x[1]) + x[2]
+        dxs = grad(fdm, f3, (x, y))
+        @test dxs[1] ≈ grad(fdm, x->f3((x, y)), x)
+        @test dxs[2] ≈ grad(fdm, y->f3((x, y)), y)
+
+        f4(d::Dict) = sum(d[:x]) + d[:y]
+        d = Dict(:x=>x, :y=>y)
+        dxs = grad(fdm, f4, d)
+        @test dxs[:x] ≈ grad(fdm, x->f3((x, y)), x)
+        @test dxs[:y] ≈ grad(fdm, y->f3((x, y)), y)
```

Comment:
another small style thing: can each of these blocks seperated by blank-lines just be there own (nested) testset? (I find testsets very helpful for maintainability)

Reply:
done.

Reply:
thank you!